### DISCUSSION TITLE
(newb question) How do I set the context for the project in VScode?

### DISCUSSION THREAD

**Discussion by Woodhouse3D** [ThePix/QuestJS#114] [Q&A] [answered]
Ok, I'm not entirely sure what its called specifically, but when I work in other engines, like Godot, the IDE automatically knows about all the functions and classes in other files in my game, alongside ones built into the engine. Unity even has some kind of function autocomplete.
Is there a setting for this?

**Comment by KVonGit** [comment]
I wonder if someone would have to create an extension for VS Code or something like that?

---
<sub>***NOTE:** This is not a suggested answer.* 🙃</sub>

**Comment by KVonGit** [comment]
Hrmm... If I hover over `createItem` in my **game/data.js** file, it only pops up a little box that says "any".

![image](https://github.com/user-attachments/assets/0988d6c5-8029-4ba9-8d0b-af209b5aa795)

---
...but, if I open **lib/_world.js**, then switch back to **game/data.js** (leaving **_world.js** open in the editor in a different tab), it works as expected when I hover over `createItem`.

![image](https://github.com/user-attachments/assets/960a5d58-736e-49d1-9f29-77f88745c99f)

**Reply by Woodhouse3D** [reply, marked as answer]
Ok, so turns out Visual Studio Code has a feature called "workspaces" which basically allows you to save your open windows as a file, so you can quickly jump back to what you were doing last time. You can set the scope as well, which has some benefits.
